lib/matplotlib/legend_handler.py

```diff
@@ -606,10 +606,8 @@ def create_artists(self, legend, orig_handle,
         leg_markerline = Line2D(xdata_marker, ydata[:len(xdata_marker)])
         self.update_prop(leg_markerline, markerline, legend)
 
-        leg_stemlines = []
-        for thisx, thisy in zip(xdata_marker, ydata):
-            l = Line2D([thisx, thisx], [bottom, thisy])
-            leg_stemlines.append(l)
+        leg_stemlines = [Line2D([thisx, thisx], [bottom, thisy])
+                         for thisx, thisy in zip(xdata_marker, ydata)]
```

Comment:
The "this" prefix doesn't add any information and just makes it harder to read. While you're at it, can you please change this to `for x, y in ...`? That would further improve the code.